gap/PolygonalComplexes/properties.gi

```diff
@@ -218,6 +218,143 @@ InstallMethod( FaceCounter, "for a twisted polygonal complex",
         return Collected( Compacted( faceDegrees ) );
     end
 );
+
+InstallMethod( ButterflyCounter, "for a simplicial surface",
+    [IsSimplicialSurface],
+    function(surface)
+	local e,temp2,counter,g,VerticesOfOrthogonalEdge,voe1,voe2,degOfVert1,degOfVert2;
+
+        VerticesOfOrthogonalEdge:=function(S,e)
+            local temp;
+            temp:=FacesOfEdge(S,e);
+            temp:=Union(VerticesOfFace(S,temp[1]),VerticesOfFace(S,temp[2]));
+            return Difference(temp, VerticesOfEdge(S,e));
+	end;
+	counter:=[];
+	for e in InnerEdges(surface) do 
+		voe1:=VerticesOfEdge(surface,e);
+		degOfVert1:=SortedList(List(voe1,g->FaceDegreeOfVertex(surface,g)));
+		voe2:=VerticesOfOrthogonalEdge(surface,e);
+		degOfVert2:=SortedList(List(voe2,g->FaceDegreeOfVertex(surface,g)));
+		Add(counter,[degOfVert1,degOfVert2]);
+	od;
+	return Set(Collected(counter));
+end
+);
+
+
+InstallMethod( UmbrellaCounter, "for a simplicial surface",
```

Comment:
The method does not work for open surfaces. I think it would be nice if also open surfaces are allowed otherwise you have to write an error for open surfaces.

Reply:
Since it is not clear how to distinguish degree lists of closed umbrellas from those of ripped umbrellas, for the meantime only closed surfaces are allowed as input.

Reply:
Okay, but you've to write this in the documentation.